src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -142,6 +143,53 @@ private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, P
             }
         }
 
+        private void ParseServerVariables(XElement serverVariables, UrlRewriteRuleBuilder builder)
+        {
+            if (serverVariables == null)
+            {
+                return;
+            }
+
+            foreach (var serverVariable in serverVariables.Elements(RewriteTags.Set))
+            {
+                ParseServerVariable(serverVariable, builder);
+            }
+        }
+
+        private void ParseServerVariable(XElement serverVariable, UrlRewriteRuleBuilder builder)
+        {
+            try
+            {
+                string prefix;
+                ServerVariableType type;
+                var name = serverVariable.Attribute(RewriteTags.Name).Value;
+
+                if (name.StartsWith(ServerVariableConstants.RequestHeaderPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
Our server variable lookup isn't this generic but probably should be.
https://github.com/aspnet/BasicMiddleware/blob/623333a919c34e63ae98a276564e03f31f9cb1cf/src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs#L25